src/components/NostoProduct/store.ts

```diff
@@ -40,41 +49,50 @@ export function createStore(element: NostoProduct) {
   }
 
   function selectSkuId(skuId: string) {
-    state.selectedSkuId = skuId
-    notify("selectedSkuId", skuId)
+    notify("selectedSkuId", (state.selectedSkuId = skuId))
+    const sku = state.skuData?.find(sku => sku.id === skuId)
+    if (sku) {
+      setSkuData(sku)
+    }
   }
 
   function selectSkuOption(optionId: string, skuIds: string[]) {
     state.skuOptions[optionId] = skuIds
     notify("skuOptions", state.skuOptions)
     const totalSelection = Object.keys(state.skuOptions).length
+    const selectedSkuIds = intersectionOf(...Object.values(state.skuOptions))
 
-    if (totalSelection === state.optionGroupCount) {
-      const selectedSkuIds = intersectionOf(...Object.values(state.skuOptions))
-      if (selectedSkuIds.length === 1) {
+    if (selectedSkuIds.length === 1) {
+      if (totalSelection === state.optionGroupCount) {
         selectSkuId(selectedSkuIds[0])
+      } else if (state.skuData) {
+        const sku = state.skuData.find(sku => sku.id === selectedSkuIds[0])
+        if (sku) {
+          setSkuData(sku)
+        }
       }
+    } else if (state.skuData) {
+      const skus = state.skuData.filter(sku => selectedSkuIds.includes(sku.id))
+      setSkuData({
+        price: getSkuValue(skus, "price"),
+        listPrice: getSkuValue(skus, "listPrice"),
+        image: getSkuValue(skus, "image"),
+        altImage: getSkuValue(skus, "altImage")
+      })
     }
   }
 
-  function setImages(image: string, altImage?: string) {
-    state.image = image
-    notify("image", image)
-
-    if (altImage) {
-      state.altImage = altImage
-      notify("altImage", altImage)
-    }
+  function setSkuData(sku: Omit<Sku, "id">) {
+    const fields = ["image", "altImage", "price", "listPrice"] as const
+    fields
+      .filter(field => sku[field])
```

Comment:
Does it work? Will the filter work for both undefined and null values? 
Also, do you think we need to check for empty string values too?

Reply:
yes
undefined, null and empty string are all falsy

Reply:
Okay. I noticed sometimes this filter gave me issues and I was forced to use filter on Boolean constructor. Maybe I got things mixed up.